### ISSUE TITLE
AWS IMDSv2 failure leads to permanent failure of ec2_instance_tags query execution

### ISSUE THREAD

**Issue by piyushml20** [osquery/osquery#8079]
# Bug report

### What operating system and version are you using?

```
osquery> SELECT version, build, platform FROM os_version;
+---------+-------+----------+
| version | build | platform |
+---------+-------+----------+
| 2       |       | amzn     |
+---------+-------+----------+
```

### What version of osquery are you using?

```
osquery> SELECT version from osquery_info;
+---------+
| version |
+---------+
| 5.4.0   |
+---------+
```

### What steps did you take to reproduce the issue?
I ran `ec2_instance_tags` query at 4 hours interval using a pack. I encountered IMDSv2 issue as noted [here](https://github.com/osquery/osquery/issues/7272). However, there are differences from that issue

1. I didn't see even a single retry call in logs
2. When I ran the same query using `osqueryi`, it's successful
3. subsequent run of `ec2_instance_tags` query will not make call to IMDSv2 and failed with same error. I thought may be token is cached but subsequent run are failing for almost a week

### What did you expect to see?
1. retry call upon failure
2. Failure when same command executed using `osqueryi`
3. subsequent run of `ec2_instance_tags` query should make IMDSv2 call

### What did you see instead?
1. I didn't see even a single retry call in logs
2. When I ran the same query using `osqueryi`, it's successful
```
osquery> SELECT instance_id, key, value FROM ec2_instance_tags;
+---------------------+-----------------------------------------------------------------------+--------------------------------------------------------------------------+
| instance_id         | key                                                                   | value                                                                    |
+---------------------+-----------------------------------------------------------------------+--------------------------------------------------------------------------+
| i-090f[REDACTED]aa862f | Name                                                                  | cluster-1-workers-eks_asg                   |
```
3. subsequent run of `ec2_instance_tags` query will not make call to IMDSv2 and failed with same error. I thought may be token is cached but subsequent run are failing for almost a week
```
Jul  7 18:49:31 ip-[REDACTED] osqueryd: I0707 18:49:31.428133  7049 scheduler.cpp:119] Executing scheduled query pack_ec2_instance_tags_snapshot: SELECT instance_id, key, value FROM ec2_instance_tags;
Jul  7 18:49:31 ip-[REDACTED] osqueryd: E0707 18:49:31.429828  7049 aws_util.cpp:223] Exception making HTTP PUT request to URL (http://169.254.169.254/latest/api/token): end of stream
Jul  7 18:49:31 ip-[REDACTED] osqueryd: I0707 18:49:31.444991  7049 ec2_instance_tags.cpp:50] Error getting EC2 instance tags: The request must contain the parameter AWSAccessKeyId

Jul  7 23:13:19 ip-[REDACTED] osqueryd: I0707 23:13:19.255720  7049 scheduler.cpp:119] Executing scheduled query pack_ec2_instance_tags_snapshot: SELECT instance_id, key, value FROM ec2_instance_tags;
Jul  7 23:13:19 ip-[REDACTED] osqueryd: I0707 23:13:19.269058  7049 ec2_instance_tags.cpp:50] Error getting EC2 instance tags: The request must contain the parameter AWSAccessKeyId
```

**Comment by Smjert** [comment]
Hello @piyushml20, could you please try the latest release?
There have been some improvements in the AWS logic that might affect you, and if anything there should also be better logging.

**Comment by piyushml20** [comment]
I tried with version 5.8.2 in debug mode. I didn't observe better logging. While executing EC2 instance metadata query it gives error along with result. can you please suggest possible cause for this error so that I can fix it:
```
osquery> select * from ec2_instance_metadata;
E0726 15:49:47.839740  9894 ec2_instance_metadata.cpp:69] Failed to get instance metadata from the metadata service
```

**Comment by piyushml20** [comment]
Thanks for the super quick response. Updated to 5.9.1 and error is gone. Original issue occured after PUT error mentioned in initial update. There is no fix time interval on when that PUT error will occur but in past I have seen that occurring after 3-4 days. So, I will share final update after few days.
Is it also possible for you to comment on separate issue I raised earlier: https://github.com/osquery/osquery/issues/8028? This issue is still present in 5.9.1.
